### PR TITLE
[5.1][Events] PageCache event classes

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

Implementing Event classes for PageCache events.
Affected events:

- [x] onPageCacheSetCaching
- [x] onPageCacheGetKey
- [x] onPageCacheIsExcluded

### Testing Instructions

Enable PageCache plugin, make sure cache works as before

### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: TBD
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by crommie** [comment]
I have tested this item :white_check_mark: successfully on fa45c91c8358d23a151b769aa060d171fc74fa24<br /><br />Caching keeps working as should.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41965">issues.joomla.org/tracker/joomla-cms/41965</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on fa45c91c8358d23a151b769aa060d171fc74fa24<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41965">issues.joomla.org/tracker/joomla-cms/41965</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41965">issues.joomla.org/tracker/joomla-cms/41965</a>.</sub>

File: plugins/system/cache/src/Extension/Cache.php [hunk 1]
```diff
@@ -148,10 +151,13 @@ public function onAfterRoute(Event $event)
             return;
         }
 
-        // If any `pagecache` plugins return false for onPageCacheSetCaching, do not use the cache.
-        PluginHelper::importPlugin('pagecache');
+        // Import "pagecache" plugins
+        $dispatcher = $this->getDispatcher();
+        PluginHelper::importPlugin('pagecache', null, true, $this->getDispatcher());
```
Comment (bembelimen) [inline plugins/system/cache/src/Extension/Cache.php]:
```suggestion
        $dispatcher = $this->getDispatcher();
        PluginHelper::importPlugin('pagecache', null, true, $dispatcher);
```

**Comment by bembelimen** [comment]
Thank you!

**Pull request merged**